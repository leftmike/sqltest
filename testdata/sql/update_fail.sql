--
-- Test UPDATE failing with primary key
--

DROP TABLE IF EXISTS tbl1;

CREATE TABLE tbl1 (c1 int primary key, c2 int, c3 int);

INSERT INTO tbl1 VALUES
    (0, 0, 0),
    (2, 2, 2),
    (4, 4, 4),
    (6, 6, 6),
    (8, 8, 8),
    (10, 10, 10);

SELECT * FROM tbl1;

UPDATE tbl1 SET c1 = c1 + 1, c2 = c2 + 1;

SELECT * FROM tbl1;

UPDATE tbl1 SET c1 = c1 - 1, c2 = c2 - 1;

SELECT * FROM tbl1;

{{Fail .Test}}
UPDATE tbl1 SET c1 = c1 + 2, c2 = c2 + 2;
