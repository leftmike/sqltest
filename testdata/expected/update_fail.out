--
-- Test UPDATE failing with primary key
--
DROP TABLE IF EXISTS tbl1;
CREATE TABLE tbl1 (c1 int primary key, c2 int, c3 int);
INSERT INTO tbl1 VALUES
    (0, 0, 0),
    (2, 2, 2),
    (4, 4, 4),
    (6, 6, 6),
    (8, 8, 8),
    (10, 10, 10);
SELECT * FROM tbl1;
   c1 c2 c3
   -- -- --
 1  0  0  0
 2 10 10 10
 3  2  2  2
 4  4  4  4
 5  6  6  6
 6  8  8  8
(6 rows)
UPDATE tbl1 SET c1 = c1 + 1, c2 = c2 + 1;
SELECT * FROM tbl1;
   c1 c2 c3
   -- -- --
 1  1  1  0
 2 11 11 10
 3  3  3  2
 4  5  5  4
 5  7  7  6
 6  9  9  8
(6 rows)
UPDATE tbl1 SET c1 = c1 - 1, c2 = c2 - 1;
SELECT * FROM tbl1;
   c1 c2 c3
   -- -- --
 1  0  0  0
 2 10 10 10
 3  2  2  2
 4  4  4  4
 5  6  6  6
 6  8  8  8
(6 rows)
{{Fail .Test}}
UPDATE tbl1 SET c1 = c1 + 2, c2 = c2 + 2;
